compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IterableAnalyzer.java

```diff
@@ -225,17 +226,18 @@ private void handleLambdaBasedIterableOperation(IterableContext context, Operati
         return supportedRetTypes;
     }
 
-    private void validateLambdaInputArgs(Operation operation, List<BType> supportedTypes, List<BType> givenTypes) {
-        if (supportedTypes.get(0).tag == TypeTags.ERROR) {
+    private void validateLambdaInputArgs(Operation operation, List<BType> actualRequiredTypes,
+                                         List<BType> userExpectedTypes) {
```

Comment:
Shall we use the terms `expectedTypes` and the `actualTypes`? Thats the convention we have used everywhere else...